### PR TITLE
WIP: Make devops component seperate from ks-installer

### PR DESCRIPTION
**Pull request by JohnNiang**
This pull request aims to make devops component separate from ks-installer. However, there are two steps we need to do:

1. Remove devops component from ks-installer(current PR is doing the right thing)
2. Add ks-devops helm charts into ks-installer(we will do it in **another PR**)

At the same time, we are doing some works for kubesphere apiserver, and the upstream branch is <https://github.com/linuxsuren/kubesphere/tree/remove-devops-ctrl>. When this branch is merged into kubesphere/kubesphere, this pull request can be ready to merge.

### What have I done

- Remove devops CRDs
- Remove devops roles files, including s2i and jenkins
- Use [ks-devops](https://github.com/LinuxSuRen/ks-devops/tree/master/charts/ks-devops) helm chart to install devops component

### Step to test

```bash
kubectl apply -f https://raw.githubusercontent.com/JohnNiang/ks-installer/962ddcef37946df0cbc34be5fe4f5ed1e0372f82/deploy/kubesphere-installer.yaml
kubectl apply -f https://raw.githubusercontent.com/JohnNiang/ks-installer/962ddcef37946df0cbc34be5fe4f5ed1e0372f82/deploy/cluster-configuration.yaml
```

/area devops
/kind optimization

### CODE REVIEW

File: deploy/kubesphere-installer.yaml [hunk 1]
```diff
@@ -264,7 +264,7 @@ spec:
       serviceAccountName: ks-installer
       containers:
       - name: installer
-        image: kubespheredev/ks-installer:latest
+        image: johnniang/ks-installer:remove-devops-comp
```
Comment (LinuxSuRen) [inline deploy/kubesphere-installer.yaml:267]: Please don't use your personal image path.

File: roles/check-result/templates/welcome.j2 [hunk 2]
```diff
@@ -7,14 +7,14 @@ Account: admin
...
+  -  After you log into the console, please check the
      monitoring status of service components in
      "Cluster Management". If any service is not
      ready, please wait patiently until all components 
      are up and running.
-  2. Please change the default password after login.
+  -  Please change the default password after login.
 {% if istio_old_version_check is defined and istio_old_version_check.stdout.find('deployed') != -1 %}
-  3. Istio has already been upraded to istio 1.6.10
+  -  Istio has already been upgraded to istio 1.6.10
```
Comment (LinuxSuRen) [inline roles/check-result/templates/welcome.j2:17]: I prefer do not change the number `1.` to `-1.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
I don't think this PR is ready to review until the branch [remove-devops-ctrl](https://github.com/linuxsuren/kubesphere/tree/remove-devops-ctrl) was merged.

/block

File: roles/check-result/templates/welcome.j2 (hunk 2 above)
Comment (JohnNiang) [reply roles/check-result/templates/welcome.j2:17]: Mee too. I'll try to fix the possiable problem as follow:

```
1. xxx
2. yyy
4. zzz
```

Comment (LinuxSuRen) [reply roles/check-result/templates/welcome.j2:17]: > I prefer do not change the number 1. to -.

Sorry about this. I prefer to do not to change it, especially in this PR. If you want to improve it, please create a new PR against it.

File: deploy/kubesphere-installer.yaml (hunk 1 above)
Comment (JohnNiang) [reply deploy/kubesphere-installer.yaml:267]: Okay, I'll provide a permanent installation link for testing.

**Comment by JohnNiang** [comment]
Due to [upstream branch](https://github.com/linuxsuren/kubesphere/tree/remove-devops-ctrl) hasn't been merged, I have to make this pr as draft again.

**Comment by JohnNiang** [comment]
I'm going to close this issue until it is ready for test running and review.

**Pull request closed without merging**